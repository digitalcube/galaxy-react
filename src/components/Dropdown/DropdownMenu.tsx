import React, { FC, PropsWithChildren, ReactNode, useState } from 'react';
import { FaEllipsisH } from 'react-icons/fa';
import {
  Dropdown as ReactstrapDropdown,
  DropdownMenu as ReactstrapDropdownMenu,
  DropdownToggle as ReactstrapDropdownToggle,
} from 'reactstrap';

export type DropdownMenuProps = PropsWithChildren<{
  right?: boolean;
  icon?: ReactNode;
  className?: string;
}>;
export const DropdownMenu: FC<DropdownMenuProps> = ({
  children,
  icon,
  right,
  className,
}) => {
  const [open, setOpen] = useState(false);
  const toggle = () => setOpen(!open);
  return (
    <ReactstrapDropdown isOpen={open} toggle={toggle} className={className}>
      <ReactstrapDropdownToggle
        className="dropdown-toggle"
        role="button"
        data-toggle="dropdown"
        aria-haspopup={true}
        aria-expanded={false}
        tag="span"
      >
        <i aria-hidden="true">{icon || <FaEllipsisH className="d-block" />}</i>
      </ReactstrapDropdownToggle>
      <ReactstrapDropdownMenu right={right}>{children}</ReactstrapDropdownMenu>
    </ReactstrapDropdown>
  );
};
