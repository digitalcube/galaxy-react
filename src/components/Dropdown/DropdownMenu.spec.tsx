/* eslint-disable jsx-a11y/anchor-is-valid */
import { render } from '@testing-library/react';
import React from 'react';
import { FaTrash } from 'react-icons/fa';
import { DropdownMenu } from './DropdownMenu';

describe('DropdownMenu', () => {
  it('to match snapshot', () => {
    const { asFragment } = render(<DropdownMenu />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('to match snapshot [full props]', () => {
    const { asFragment } = render(
      <DropdownMenu
        right
        icon={<FaTrash className="d-block" />}
        className="test"
      >
        <a href="#">Item</a>
      </DropdownMenu>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
