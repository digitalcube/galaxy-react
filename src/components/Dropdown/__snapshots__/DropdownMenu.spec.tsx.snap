// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownMenu to match snapshot [full props] 1`] = `
<DocumentFragment>
  <div
    class="test dropdown"
  >
    <span
      aria-expanded="false"
      aria-haspopup="true"
      class="dropdown-toggle"
      data-toggle="dropdown"
      role="button"
    >
      <i
        aria-hidden="true"
      >
        <svg
          class="d-block"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
          />
        </svg>
      </i>
    </span>
    <div
      aria-hidden="true"
      class="dropdown-menu dropdown-menu-right"
      role="menu"
      tabindex="-1"
    >
      <a
        href="#"
      >
        Item
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DropdownMenu to match snapshot 1`] = `
<DocumentFragment>
  <div
    class="dropdown"
  >
    <span
      aria-expanded="false"
      aria-haspopup="true"
      class="dropdown-toggle"
      data-toggle="dropdown"
      role="button"
    >
      <i
        aria-hidden="true"
      >
        <svg
          class="d-block"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M328 256c0 39.8-32.2 72-72 72s-72-32.2-72-72 32.2-72 72-72 72 32.2 72 72zm104-72c-39.8 0-72 32.2-72 72s32.2 72 72 72 72-32.2 72-72-32.2-72-72-72zm-352 0c-39.8 0-72 32.2-72 72s32.2 72 72 72 72-32.2 72-72-32.2-72-72-72z"
          />
        </svg>
      </i>
    </span>
    <div
      aria-hidden="true"
      class="dropdown-menu"
      role="menu"
      tabindex="-1"
    />
  </div>
</DocumentFragment>
`;
